python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2390,6 +2399,38 @@ def get_handlers(self) -> Sequence[logging.Handler]:
     def store_event(self, event: "EventLogEntry") -> None:
         self._event_storage.store_event(event)
 
+    def _should_retry_run(self, run: DagsterRun) -> bool:
+        from dagster._core.events import RunFailureReason
+
+        if run.status == DagsterRunStatus.FAILURE and self.run_retries_enabled:
+            max_retries = (
+                int(run.tags[MAX_RETRIES_TAG])
+                if run.tags.get(MAX_RETRIES_TAG) is not None
+                else self.run_retries_max_retries
+            )
```

Comment:
Apologies in advance for pedanticness; I don't love implicit int coercion - I find it confusing, I always think it's more sensible to leave None values as None if they're not set. The following line
`if max_retries > 0:`
almost makes it seems like it's valid to set `max_retries` to 0, when in fact 0 is the null state. I just think it's always better to be explicit

Reply:
i'm not sure i follow, what do you suggest instead?

Reply:
@dpeng817  I think it is valid to set max_retries to 0 - for example if the deployment-wide default is 2, but you don't want a particular run or job to retry

Reply:
Ah I completely misread this actually somehow. My brain interpreted this as coercion of an optional. Disregard, apologies